firedrake/assemble.py

```diff
@@ -425,10 +426,31 @@ def allocate_matrix(expr, bcs=None, *, mat_type=None, sub_mat_type=None,
                                          for get_map, regions in domains.items()
                                          if regions))
     try:
+        # Let tensor=None and bcs=None for the purpose of allocation.
```

Comment:
This comment is a bit obscure. I understand why `tensor` must be `None`, but why `bcs`?

Reply:
`bcs` has not been used to construct sparsity. This allows for reusing the constructed sparsity for the same problem with slightly different BCs.

Reply:
Sorry, my feedback was a bit obscure. Could you reword the comment to make it clearer?

Reply:
Done.